src/main/kotlin/org/opensearch/replication/action/stop/TransportUnfollowIndexReplicationAction.kt

```diff
@@ -0,0 +1,68 @@
+import org.opensearch.common.inject.Inject
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver
+import org.opensearch.cluster.service.ClusterService
+import org.opensearch.commons.replication.action.ReplicationActions.STOP_REPLICATION_ACTION_TYPE
+import org.opensearch.commons.replication.action.StopIndexReplicationRequest
+import org.opensearch.commons.replication.action.ReplicationActions.UNFOLLOW_REPLICATION_ACTION_NAME
+import org.opensearch.commons.utils.recreateObject
+import org.opensearch.core.action.ActionListener
+import org.opensearch.replication.metadata.ReplicationMetadataManager
+import org.opensearch.replication.util.coroutineContext
+import org.opensearch.replication.util.stackTraceToString
+import org.opensearch.replication.util.suspendExecute
+import org.opensearch.tasks.Task
+import org.opensearch.threadpool.ThreadPool
+import org.opensearch.transport.TransportService
+
+
+/**
+ * This action transforms the request from ActionRequest type to StopIndexReplicationRequest
+ * and performs the TransportStopIndexReplicationAction on it.
+ * While TransportStopIndexReplicationAction is used directly by the _stop replication REST API,
+ * this action is used for inter-plugin communication by ism plugin to unfollow i.e. stop replication.
+ */
+class TransportUnfollowIndexReplicationAction @Inject constructor (
+    val name: String,
+    val transportService: TransportService,
+    val clusterService: ClusterService,
+    val threadPool: ThreadPool,
+    val client: Client,
+    val actionFilters: ActionFilters,
+    val indexNameExpressionResolver: IndexNameExpressionResolver,
+    val replicationMetadataManager: ReplicationMetadataManager,
+): HandledTransportAction<ActionRequest, AcknowledgedResponse> (UNFOLLOW_REPLICATION_ACTION_NAME, transportService, actionFilters, ::StopIndexReplicationRequest),
+    CoroutineScope by GlobalScope  {
+    companion object {
+        private val log = LogManager.getLogger(TransportUnfollowIndexReplicationAction::class.java)
+    }
+
+    @Throws(Exception::class)
+    override fun doExecute(task: Task?, request: ActionRequest, listener: ActionListener<AcknowledgedResponse>?) {
+        launch(Dispatchers.Unconfined + threadPool.coroutineContext()) {
+            val transformedRequest = request as? StopIndexReplicationRequest
+                ?: request.let { recreateObject(it) { StopIndexReplicationRequest(it) } }
+            try {
+
+                var response = client.suspendExecute(STOP_REPLICATION_ACTION_TYPE, transformedRequest, true)
+                log.info("Stop replication successful for index[${transformedRequest.indexName}] with response: " + response.isAcknowledged)
+                listener?.onResponse(AcknowledgedResponse(true))
+            } catch (e: Exception) {
+                log.error("Stop replication failed for index[${transformedRequest.indexName}] with error ${e.stackTraceToString()}")
+                listener?.onFailure(e)
+                throw e

```

Comment:
Probably no need to rethrow the exception here

Reply:
Yes, you are right. I had corrected that change in my local, but missed pushing it. My bad!